content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx

```diff
@@ -901,6 +901,59 @@ enum Role {
 
 ## Defining composite types
 
+<Admonition type="info">
+
+Composite types are available in Preview in versions `3.10.0` and later.
+
+</Admonition>
+
+<Admonition type="warning">
+
+Composite types are currently only available on MongoDB.
+
+</Admonition>
+
+Composite types (known as [embedded documents](https://docs.mongodb.com/manual/core/data-model-design/#std-label-data-modeling-embedding) in MongoDB) provide support for embedding records inside other records, by allowing you to define new object types. Composite types are structured and typed in a similar way to models.
+
+To use composite types with MongoDB, you will need to enable the `mongoDb` Preview feature in your `schema.prisma` file:
+
+```prisma file="schema.prisma"
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["mongoDb"]
+}
+```
+
+To define a composite type, use the `type` block. As an example, take the following schema:
+
+```prisma file=schema.prisma
+model Product {
+id     String  @id
+name   String
+photos Photo[]
+}
+
+type Photo {
+height Int
+width  Int
+url    String
+}
+```
+
+In this case, the `Product` model has an embedded `Photo[]` type.
+
+### Considerations when using composite types:
+
+Composite types currently have some limitations:
+
+- There is limited support for [attributes](/reference/api-reference/prisma-schema-reference#attributes) within composite types. Only `@default`, `@map` and native types (e.g. `@db.ObjectId`) are supported. The `@unique`, `@id`, `@relation`, `@ignore`, `@updatedAt` and `@createdAt` attributes are not supported.
```

Comment:
- Is this really a limitation or just a difference to models and how they work?
- Are native types really not supported?
- `@relation` is covered by the list item below.

Reply:
> Are native types really not supported?

Might be a bit wordy, but they are supported. It says: _Only `@default`, `@map` and native types (e.g. `@db.ObjectId`) are supported_

> Is this really a limitation or just a difference to models and how they work?

Some work differently, some could be supported in the future.

Reply:
I've made this into a list form instead - is this any clearer?

Reply:
Looks good on my end!